src/main/java/net/redstonecraft/lanservers/LanServers.java

```
@@ -24,18 +31,55 @@ public void enable() {
-                } catch (IOException e) {
-                    e.printStackTrace();
+                for (MulticastSocket socket :  sources) {
+                    try {
+                        socket.send(packet);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
                 }
             }
         }, 0, 1500);
     }
 
     public void disable() {
         timer.cancel();
-        socket.close();
+        for (MulticastSocket socket :  sources) {
+            socket.close();
+        }
     }
 
+    private static List<MulticastSocket> getSourceSockets() {
+        try {
+            // Take all network interfaces
+            return Collections.list(NetworkInterface.getNetworkInterfaces()).stream()
+                    // keep all relevant (not localhost, up, can multicast)
+                    .filter(iface -> {
+                        try {
+                            return iface.supportsMulticast() && iface.isUp() && !iface.isLoopback();
+                        } catch (SocketException e) {
+                            return false;
+                        }
+                    })
+                    // interface can have multiple addresses, make a loooong list
+                    .flatMap(iface -> iface.getInterfaceAddresses().stream()
+                            // get actual address
+                            // make socket, ignore errors
+                            .map(ifaddr -> ifaddr.getAddress())
+                            .filter(inaddr -> (inaddr instanceof Inet4Address))
+                            .map(inaddr -> new InetSocketAddress(inaddr, multicastPort))
+                            .map(insaddr -> {
+                                try {
+                                    return new MulticastSocket(insaddr);
+                                } catch (IOException e) {
+                                    return null;

```

No stacktrace is printed and no comment describing a reason for silencing the error.

---

I make a comment

---

There is no comment in the catch block describing the reason.